websocket_push_session.go

```diff
@@ -229,8 +256,32 @@ func (s *wsPushSession) listen() {
 
 			pushConfig := elemental.NewPushConfig()
 			if err := elemental.Decode(s.encodingRead, data, pushConfig); err != nil {
-				s.close(websocket.CloseUnsupportedData)
-				return
+				if !s.handlesErrorEvents() {
+					s.close(websocket.CloseUnsupportedData)
+					return
+				}
+
+				s.setErrorState(true)
+				msgpack, json, err := prepareEventData(elemental.NewErrorEvent(elemental.Error{
+					Description: fmt.Sprintf("could not decode message into %T: %s", pushConfig, err),
+					Subject:     "bahamut",
+					Title:       "Bad request",
+				}, s.encodingWrite))
+
+				if err != nil {
+					zap.L().Error("unable to prepare error event encoding, closing socket", zap.Error(err))
+					s.close(websocket.CloseInternalServerErr)
+					return
+				}
```

Comment:
@primalmotion Can't think of a way to test this codepath...it's unreachable.

Reply:
unless there is a major screw up in elemental, yeah it's not reachable. it's ok

Reply:
Precisely 👌

Reply:
Note: I raised this because codecov is mad 😅